subsys/net/l2/wifi/wifi_nm.c

```diff
@@ -10,30 +10,39 @@ LOG_MODULE_REGISTER(wifi_nm, CONFIG_WIFI_NM_LOG_LEVEL);
 #include <zephyr/kernel.h>
 #include <zephyr/net/wifi_nm.h>
 
+/* Used to protect nm data */
+static K_MUTEX_DEFINE(wifi_nm_lock);
+
 struct wifi_nm_instance *wifi_nm_get_instance(const char *name)
 {
+	k_mutex_lock(&wifi_nm_lock, K_FOREVER);
```

Comment:
why do we need a lock here? this isn't tied any interface, so, can be used at any time lock free.